src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/MarshalTests.cs

```diff
@@ -152,5 +152,134 @@ public static void GetHRForException()
                 Marshal.GetExceptionForHR(e.HResult);
             }
         }
+
+        [Fact]
+        public static void GenerateGuidForType()
+        {
+             Assert.Throws<PlatformNotSupportedException>(() => Marshal.GenerateGuidForType(typeof(System.Int32)));        
```

Comment:
In the excel sheet for interop,  it asked to throw for CoreCLR. It should be easy enough to enable it for windows. I suppose I can check for PNSE for non-windows here.